src/main/java/frc/robot/subsystems/ShooterSubsystem.java

```diff
@@ -2,21 +2,24 @@
 
 import com.revrobotics.CANPIDController;
 import com.revrobotics.CANSparkMax;
-import com.revrobotics.ControlType;
-import com.revrobotics.CANSparkMaxLowLevel.MotorType;
-
 import edu.wpi.first.wpilibj.Solenoid;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
-import static frc.robot.Constants.*;
 
 public class ShooterSubsystem extends SubsystemBase {
-    private CANSparkMax shooterMotor1 = new CANSparkMax(SHOOTER_MOTOR_1, MotorType.kBrushed);
-    private CANSparkMax shooterMotor2 = new CANSparkMax(SHOOTER_MOTOR_2, MotorType.kBrushed);
-    private Solenoid shooterSolenoid = new Solenoid(0);
-    private CANPIDController shooterController = shooterMotor1.getPIDController();
 
-    
-    public ShooterSubsystem() {
+    private final CANSparkMax shooterMotor1;
+    private final CANSparkMax shooterMotor2;
+    private final Solenoid shooterSolenoid;
+    private final CANPIDController shooterController;
+
+    public ShooterSubsystem(CANSparkMax shooterMotor1,
+                            CANSparkMax shooterMotor2,
+                            Solenoid shooterSolenoid) {
```

Comment:
Notice how the subsystems don't create their own components. This makes testing easier, because we can give them fake components that we can monitor